src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -1,268 +1,66 @@
-import BackButton from '../back-button';
-import NextButton from '../next-button';
-
-import AddRecordingSetButton from './add-recording-set-button';
-import CreatedRecordingSetList from './created-recording-set-list';
-import SetMetaConfiguration from './set-meta-configuration';
-import SetRecordingListConfiguration from './set-recording-list-configuration';
-import { BuiltInRecordingList } from './types';
+import { Consumer, RecordingSet } from '../../types';
 
+import ConfigureRecordingSet from './configure-recording-set';
 import './show-details.scss';
+import Dvcfg from './dvcfg';
+import ListPreview from './list-preview';
+import OtoIni from './oto-ini';
 
-interface ProjectFile extends RecordingProject {
-  name: string;
-  recordingSets: RecordingSet[];
-}
-
-const DUMMY_PROJECT_FILE = {
-  name: '',
-  rootPath: '',
-  recordingSets: [],
-};
-
-const BUILT_IN_RECORDING_LISTS: BuiltInRecordingList[] = [
-  'デルタ式英語リストver5 (Delta English Ver. 5)',
-  'Z式CVVC-Normal (Z Chinese CVVC - Normal)',
-];
+export type RecordingPageState = 'home' | 'list-preview' | 'oto-ini' | 'dvcfg' | 'external';
 
 const ConfigureRecordingSetPage: FC<{
   onNext: MouseEventHandler<HTMLElement>;
   onBack: MouseEventHandler<HTMLElement>;
   onSetSelected: Consumer<RecordingSet>;
 }> = ({ onNext, onBack, onSetSelected }) => {
-  const { t } = useTranslation();
-  const { recordingProject } = useContext(RecordingProjectContext);
-  const [projectFile = DUMMY_PROJECT_FILE, setProjectFile] = useLocalStorage<ProjectFile>(
-    getLSKey('ConfigureRecordingSetPage', 'projectFile'),
-    DUMMY_PROJECT_FILE,
-  );
-  const [recordingSets = [], setRecordingSets] = useLocalStorage<RecordingSet[]>(
-    getLSKey('ConfigureRecordingSetPage', 'recordingSets'),
-    [],
-  );
-  const [showingDetails = false, setShowingDetails] = useLocalStorage(
-    getLSKey('ConfigureRecordingSetPage', 'showingDetails'),
-    false,
-  );
-  const [canWrite, setCanWrite] = useState(false);
-
-  useEffect(() => {
-    if (recordingProject) {
-      (async (): Promise<void> => {
-        const { rootPath } = recordingProject;
-        const configFilePath = join(rootPath, PROJECT_CONFIG_FILENAME);
-
-        const fileExistence = await checkFileExistence(configFilePath);
-        if (!fileExistence) {
-          setProjectFile(DUMMY_PROJECT_FILE);
-        } else if (fileExistence === 'folder') {
-          throw new Error(`Config file cannot be read because it is a directory: ${configFilePath}`);
-        } else {
-          // TODO Write a loader
-          const projectState = JSON.parse(await readFile(configFilePath)) as ProjectFile;
-          setProjectFile(projectState);
-          setRecordingSets(projectState.recordingSets);
-        }
-        setCanWrite(true);
-      })();
+  const [recordingSetState, setRecordingSetState] = useState<RecordingPageState>('home');
+  const [prevRecordingSetState, setPrevRecordingSetState] = useState<RecordingPageState>('external');
```

Comment:
Is this the previous state of `recordingSetState`? Have you looked at `usePrevious`?
https://github.com/streamich/react-use/blob/master/docs/usePrevious.md